wire/msgtx.go

```diff
@@ -326,6 +328,33 @@ func (t TxWitness) ToHexStrings() []string {
 	return result
 }
 
+// MarshalJSON serialises the structure as a JSON appropriate value.
+//
+// NOTE: The 2D byte array is encoded as an array of hex strings.
+func (t TxWitness) MarshalJSON() ([]byte, error) {
```

Comment:
Do we really want those `MarshalJSON`/`UnmarshalJSON` methods? Those represent a breaking change for users that might have relied on the "old" serialization format. Didn't we have the same problem with the chain hash where adding the (un)marshal methods broke some user's code?
I don't really like Go's approach to JSON serialization because depending on a method being present or not makes the serialization behavior quite non-deterministic when updating dependencies.
That's why I think we shouldn't rely on it (which is probably also the reason those methods don't exist on most of our structs).

Reply:
Will remove from PR.